File: SpringBootAPI/src/main/java/com/training/pokedex/entity/Pokemon.java
package com.training.pokedex.entity;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

@Entity
public class Pokemon {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public long id;
    public String name;
    public String type1;
    public String type2;
    public String img_url;
    public Long evolves_to;
    public Long evolves_from;

    public Pokemon() {
    }

    public Pokemon(long id, String name, String type1, String type2, String img_url, long evolves_to,
            long evolves_from) {
        this.id = id;
        this.name = name;
        this.type1 = type1;
        this.type2 = type2;
        this.img_url = img_url;
        this.evolves_to = evolves_to;
        this.evolves_from = evolves_from;
    }

}

File: AngularFrontend/pokedex/src/app/Entity/pokemon.ts
export class Pokemon {
  name: string;
  id: number;
  type1: PokeType;
  type2: PokeType;
  img_url: string;
  evolves_to: number;
  evolves_from: number;

  constructor(
    name: string,
    id: number,
    type1: PokeType,
    type2: PokeType,
    img_url: string,
    evolves_to: number,
    evolves_from: number
  ) {
    this.name = name;
    this.id = id;
    this.type1 = type1;
    this.type2 = type2;
    this.img_url = img_url;
    this.evolves_to = evolves_to;
    this.evolves_from = evolves_from;
  }
}

export enum PokeType {
  Acier = 'ACIER',
  Combat = 'COMBAT',
  Dragon = 'DRAGON',
  Eau = 'EAU',
  Electrik = 'ELECTRIK',
  Fee = 'FEE',
  Feu = 'FEU',
  Glace = 'GLACE',
  Insecte = 'INSECTE',
  Normal = 'NORMAL',
  Plante = 'PLANTE',
  Poison = 'POISON',
  Psy = 'PSY',
  Roche = 'ROCHE',
  Sol = 'SOL',
  Spectre = 'SPECTRE',
  Tenebres = 'TENEBRES',
  Vol = 'VOL',
}
